core/trino-main/src/test/java/io/trino/execution/TestNodeScheduler.java

```diff
@@ -333,9 +340,27 @@ public void testMaxSplitsPerNode()
         assertEquals(nodeTaskMap.getPartitionedSplitsOnNode(newNode), 0);
     }
 
+    @Test
+    public void testBasicAssignmentMaxUnacknowledgedSplitsPerTask()
+    {
+        setUpNodes();
+        // One split for each node, and one extra split that can't be placed
+        int splitCount = nodeManager.getActiveConnectorNodes(CONNECTOR_ID).size() + 1;
+        Set<Split> splits = new HashSet<>();
+        for (int i = 0; i < splitCount; i++) {
+            splits.add(new Split(CONNECTOR_ID, new TestSplitRemote(), Lifespan.taskWide()));
+        }
+        Multimap<InternalNode, Split> assignments = nodeSelector.computeAssignments(splits, ImmutableList.copyOf(taskMap.values()), 1).getAssignments();
+        assertEquals(assignments.entries().size(), 3);
```

Comment:
Why use the constant 3 here and `nodeManager.getActiveConnectorNodes(CONNECTOR_ID).size()` above?

Reply:
Extracted common variable for node count to use here.